### PR TITLE
add get default method

### LINKED ISSUE #382: SchemaField#default: provide variant that decodes the document internally
Something like 

```
//in implicit class for SchemaField[S, A]
def getDefaultValue: Option[A] = getDefault.map(getCachedDocumentDecoder[A].apply(_))
```

### PR DESCRIPTION
**Pull request by lewisjkl**
Adds methods to get a schema's default (from the default hint) and to decode the document from the hint with `getDefaultValue`.

Closes #382.

### CODE REVIEW

**Comment by lewisjkl** [comment]
Converting to draft until 0.17 CI issues are worked out

File: modules/core/src/smithy4s/schema/Schema.scala [hunk 1]
```diff
@@ -78,6 +78,11 @@ sealed trait Schema[A]{
 
   final def refined[B]: PartiallyAppliedRefinement[A, B] = new PartiallyAppliedRefinement[A, B](this)
 
+  final def getDefault: Option[Document] =
+      this.hints.get(smithy.api.Default).map(_.value)
+
+  final def getDefaultValue: Option[A] = getDefault.flatMap(Document.Decoder.fromSchema(this).decode(_).toOption)
```
[the remark this answers is not in the document]
Comment (lewisjkl) [reply modules/core/src/smithy4s/schema/Schema.scala:84]: hmm that's a good point, I'm not sure. I can't immediately see an issue with using a lazy val instead, but will defer to what Oli has to say about it.

**Review by kubukoz — APPROVED** [review]

**Review by daddykotex — APPROVED** [review]

**Comment by daddykotex** [comment]
I believe this is ready to go down following your merge of `series/0.17`

File: modules/core/src/smithy4s/schema/Schema.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (Baccata) [reply modules/core/src/smithy4s/schema/Schema.scala:84]: I think a def is fine.

**Comment by Baccata** [comment]
It'd have been nice to actually use the method where defaults are needed

**Comment by lewisjkl** [comment]
> It'd have been nice to actually use the method where defaults are needed

Will do this in another PR

**Pull request merged**